pkg/policy/rule.go

```diff
@@ -27,6 +29,19 @@ import (
 
 type rule struct {
 	api.Rule
+
+	// mutex protects everything below it (i.e., not the rule itself).
+	mutex lock.RWMutex
+
+	// localRuleConsumers is the set of the numeric identifiers which this rule
+	// selects which are node-local (e.g., Endpoint).
+	localRuleConsumers map[uint16]*identity.Identity
```

Comment:
1. Rename this.
2. Group with processedConsumers in a single type to signify that they are 'closed' and related.